### PR TITLE
[016][P5][Billing] Implement pending meter, meter history, and financial reports

### LINKED ISSUE #19: [016][P5][Billing][Read+Command] Implement pending-meter, meter-history, and financial report APIs
## Goal
Implement billing read models for pending meter work, meter history, and financial reports, plus financial report send.

## Scope
- Implement `GET /properties/{id}/pending-meter`
- Implement `GET /properties/{id}/meter-history`
- Implement `GET /rooms/{id}/meter-history`
- Implement `GET /properties/{id}/financial-report`
- Implement `GET /properties/{id}/financial-report/{year}/{month}`
- Implement `POST /properties/{id}/financial-report/{year}/{month}/send`
- Make pending-meter and history/report reads consistent with cadence-driven bill periods

## Out of Scope
- Bill detail/meter/payment core APIs
- Scheduler monthly snapshot generation implementation
- Notification management APIs

## API Contract
- Endpoints above with roles, filters, and response shapes per OpenAPI
- Error codes include `FORBIDDEN`, `PROPERTY_NOT_FOUND`, and general server/query failures

## Data Impact
- Read tables: `bills`, `monthly_snapshots`, `monthly_snapshot_entries`, `property_accounts`, `accounting_entries`, `properties`, `rooms`
- Write effects: financial report send may trigger notification send or output dispatch but should not alter unrelated billing state
- Query type: read model and report aggregation

## Business Rules
- Pending-meter list must only include `pending_meter` bills.
- Pending-meter, meter-history, and bill-detail-oriented report data must respect each bill's `period_start` and `period_end` rather than assuming every bill is a single calendar month.
- Meter history must reflect room/property billing history across monthly and bimonthly cadence.
- Financial report reads must combine current-month and historical snapshot data as designed.
- Financial report send is admin/organizer review/send action only.

## Authorization
- Valid Firebase token required
- Property access required
- Send action restricted per OpenAPI roles

## Domain Mapping
- BC: Billing
- Query: pending meter, meter history, report summary/detail
- Use case: send financial report
- Aggregate/read model: Billing report query models
- Produced events: if send action emits/uses notification trigger, document it in code
- Consumed events: none in this ticket

## Implementation
- Handler: read/report/send endpoints
- Query repository: pending meter, history, and report aggregation
- Query semantics: expose cadence-driven periods accurately instead of flattening them into assumed monthly buckets
- Send action: report dispatch coordination and response handling

## References
- `docs/design/domain-model.md`
- `docs/spec/src/paths/billing/properties_{id}_pending-meter.yaml`
- `docs/spec/src/paths/billing/properties_{id}_meter-history.yaml`
- `docs/spec/src/paths/billing/rooms_{id}_meter-history.yaml`
- `docs/spec/src/paths/billing/properties_{id}_financial-report.yaml`
- `docs/spec/src/paths/billing/properties_{id}_financial-report_{year}_{month}.yaml`
- `docs/spec/src/paths/billing/properties_{id}_financial-report_{year}_{month}_send.yaml`

## Acceptance Criteria
- [ ] Pending-meter read works
- [ ] Property and room meter history reads work
- [ ] Pending-meter and meter-history outputs preserve correct bill-period boundaries for cadence-driven bills
- [ ] Financial report summary/detail reads work
- [ ] Financial report send works for allowed roles
- [ ] Responses match OpenAPI

## Tests
- Pending-meter happy path
- Pending-meter with non-monthly bill period
- Meter history happy path
- Meter history across mixed cadence periods
- Financial report summary/detail happy path
- Empty-report cases
- Financial report send authorization and success path

## Definition of Done
- [ ] Billing report/query APIs implemented
- [ ] Financial report send implemented
- [ ] Cadence-aware bill-period semantics are preserved in pending-meter/history/report queries
- [ ] OpenAPI remains aligned
- [ ] Tests cover read/report/send behavior

### PR DESCRIPTION
**Pull request by ifan0927**
## Summary

- Implement billing pending-meter and meter-history read APIs with cadence-aware bill period semantics.
- Implement financial report summary/detail reads across live current-month accounting entries and historical monthly snapshots.
- Implement financial report send as a Billing action that validates the report and emits `FinancialReportSendRequested` for downstream notification handling.
- Add focused application, repository, handler, and router tests for response shapes, authorization policy, period overlap filtering, empty report handling, and event publication.

## Notes

- Notification subscriber implementation is intentionally deferred and tracked in #37.
- `PropertyCreated -> PropertyAccount` subscriber remains tracked in #37 and is not folded into this issue.

Closes #19

## Validation

- `env GOCACHE=/tmp/go-cache go test ./...`
- `git diff --check`

### CODE REVIEW

File: internal/application/billing/reports_test.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+		t.Fatalf("expected 1 event, got %d", len(publisher.events))
+	}
+	event, ok := publisher.events[0].(domainevents.FinancialReportSendRequested)
+	if !ok {
+		t.Fatalf("expected FinancialReportSendRequested, got %T", publisher.events[0])
+	}
+	if event.PropertyID != testPropertyID || event.Year != 2026 || event.Month != 3 || !event.OccurredAt.Equal(occurredAt) {
+		t.Fatalf("unexpected event: %+v", event)
+	}
+}
```
[the remark this answers is not in the document]
Comment (ifan0927) [reply internal/application/billing/reports_test.go:314]: Addressed in `44ac6d8`.

Added `TestSendFinancialReportUsesCurrentMonthLivePath`, which fixes the clock to the target month, verifies the service uses `liveReportQuery` rather than the snapshot path, and still asserts that `FinancialReportSendRequested` is published with the expected property/year/month/occurred_at payload.

File: internal/application/billing/reports.go [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+
+// NewSendFinancialReportService returns a SendFinancialReportService.
+func NewSendFinancialReportService(repo ReportRepository, publisher domainevents.Publisher, clock Clock) *SendFinancialReportService {
+	if publisher == nil {
+		publisher = domainevents.NoopPublisher{}
+	}
+	if clock == nil {
+		clock = systemClock{}
+	}
+	return &SendFinancialReportService{repo: repo, publisher: publisher, clock: clock}
```
[the remark this answers is not in the document]
Comment (ifan0927) [reply internal/application/billing/reports.go:413]: Addressed in `44ac6d8`.

Removed the silent `NoopPublisher` substitution from `NewSendFinancialReportService`. `Execute` now returns an internal error if the service is constructed without a publisher, so missing event wiring is no longer treated as a successful send. Added `TestSendFinancialReportRequiresPublisher` for this behavior.
